elements/src/main/java/org/vaadin/elements/elementui.js

```diff
@@ -33,7 +33,10 @@ window.org_vaadin_elements_ElementIntegration = function() {
 						var attributesToUpdate = boundAttributes[id][event];
 						for (var i = 0; i < attributesToUpdate.length; i++) {
 							var attribute = attributesToUpdate[i];
-							attributeChange = [+id, attribute, ""+ids[id][attribute]];
+							var valueToSend = ids[id][attribute];
+							attributeChange = [+id, attribute, valueToSend != null && typeof valueToSend === 'object' ?
+								JSON.stringify(valueToSend) : "" + valueToSend];
```

Comment:
Is the original  `"" + valueToSend` even necessary, i.e. is there any reasonable case where `JSON.stringify(valueToSend)` on its own wouldn't suffice?

Reply:
Makes the usage for simple properties easier, for instance Strings or numbers. In my tests these have been wrapped in extra quotes and needed also extra parsing. Could be a bigger breaking change, if these cases would have to been rewritten, too (instead of only complex value cases, which are maybe not so often the case).